test/test_pub_sub.py.in

```diff
@@ -0,0 +1,307 @@
+                                         node_main_test):
+        self.addCleanup(_cleanUpLogs, performance_log_prefix_pub)
+        self.addCleanup(_cleanUpLogs, performance_log_prefix_sub)
+
+        launch_testing.asserts.assertExitCodes(
+            proc_info,
+            [launch_testing.asserts.EXIT_OK],
+            node_relay_test,
+        )
+
+        launch_testing.asserts.assertExitCodes(
+            proc_info,
+            [launch_testing.asserts.EXIT_OK],
+            node_main_test,
+        )
+
+        performance_logs = glob(performance_log_prefix_cpumem_pub + '*')
+        performance_logs_perf_test = glob(performance_log_prefix_pub + '_*')
+
+        if performance_logs and performance_logs_perf_test:
+            performance_report_csv = os.environ.get('PERFORMANCE_OVERHEAD_CSV')
+            if performance_report_csv:
+                _raw_to_csv(performance_logs[0], performance_logs_perf_test[0],
+                            performance_report_csv, "pub")
+            else:
+                print('No CSV report written - set PERFORMANCE_OVERHEAD_CSV to write a report',
+                      file=sys.stderr)
+
+            performance_report_png = os.environ.get('PERFORMANCE_OVERHEAD_PNG')
+            if performance_report_png:
+                _raw_to_png(performance_logs[0], performance_logs_perf_test[0],
+                            performance_report_png, "publisher", "@COMM_PUB@")
+            else:
+                print('No PNG report written - set PERFORMANCE_OVERHEAD_PNG to write a report',
+                      file=sys.stderr)
+        else:
+            print('No report written - no performance log was produced', file=sys.stderr)
+
+        performance_logs = glob(performance_log_prefix_cpumem_sub + '*')
+        performance_logs_perf_test = glob(performance_log_prefix_sub + '_*')
+
+        if performance_logs and performance_logs_perf_test:
+            performance_report_csv = os.environ.get('PERFORMANCE_OVERHEAD_CSV')
+            if performance_report_csv:
+                _raw_to_csv(performance_logs[0], performance_logs_perf_test[0],
+                            performance_report_csv, "sub")
+            else:
+                print('No CSV report written - set PERFORMANCE_OVERHEAD_CSV to write a report',
+                      file=sys.stderr)
+
+            performance_report_png = os.environ.get('PERFORMANCE_OVERHEAD_PNG')
+            if performance_report_png:
+                _raw_to_png(performance_logs[0], performance_logs_perf_test[0],
+                            performance_report_png, "subscriber", "@COMM_SUB@")
+            else:
+                print('No PNG report written - set PERFORMANCE_OVERHEAD_PNG to write a report',
+                      file=sys.stderr)
+        else:
+            print('No report written - no performance log was produced', file=sys.stderr)
+
+        self.addCleanup(_cleanUpLogs, performance_log_prefix_cpumem_pub)
```

Comment:
This should happen right away - this function doesn't actually remove the logs, it just adds them to the list of logs to be cleaned up when the test finishes.

The risk of doing it here (and the reason I didn't initially just `os.remove` the logs at the end) is that if the test fails and the function returns prematurely, the logs get left behind.